apps/api/src/app/vercel-integration/usecases/setup-integration/setup-integration.usecase.ts

```diff
@@ -0,0 +1,162 @@
+import { SetupIntegrationResponseDto } from '../../dtos/setup-integration-response.dto';
+import { SetupIntegrationCommand } from './setup-integration.command';
+
+interface ISetEnvironment {
+  token: string;
+  projectIds: string[];
+  teamId: string;
+  identifier: string;
+  secretKey: string;
+}
+
+@Injectable()
+export class SetupIntegration {
+  constructor(
+    private httpService: HttpService,
+    private getApiKeyUsecase: GetApiKeys,
+    private getEnvironmentUsecase: GetEnvironment
+  ) {}
+
+  async execute(command: SetupIntegrationCommand): Promise<SetupIntegrationResponseDto> {
+    try {
+      const tokenData = await this.getVercelToken(command.vercelIntegrationCode);
+
+      const projects = await this.getVercelProjects(tokenData.accessToken, tokenData.teamId);
+
+      const projectIds = projects.map((project) => project.id);
+
+      const envKeys = await this.getEnvKeys(command);
+
+      await this.setEnvironments({
+        projectIds,
+        identifier: envKeys.clientKey,
+        secretKey: envKeys.privateKey,
+        teamId: tokenData.teamId,
+        token: tokenData.accessToken,
+      });
+
+      return {
+        success: true,
+      };
+    } catch (error) {
+      throw new ApiException(
+        error?.response?.data?.error_description || error?.response?.data?.message || error.message
+      );
+    }
+  }
+
+  async getVercelToken(code: string): Promise<{
+    accessToken: string;
+    userId: string;
+    teamId: string;
+  }> {
+    const postData = new URLSearchParams({
+      code,
+      client_id: process.env.VERCEL_CLIENT_ID,
+      client_secret: process.env.VERCEL_CLIENT_SECRET,
+      redirect_uri: process.env.VERCEL_REDIRECT_URI,
+    });
+
+    const response = await lastValueFrom(
+      this.httpService.post(`${process.env.VERCEL_BASE_URL}/v2/oauth/access_token`, postData, {
+        headers: {
+          'Content-Type': 'application/x-www-form-urlencoded',
+        },
+      })
+    );
+
+    const data = await response.data;
+
+    return {
+      accessToken: data.access_token,
+      userId: data.user_id,
+      teamId: data.team_id,
+    };
+  }
+
+  async getVercelProjects(token: string, teamId: string) {
+    const response = await lastValueFrom(
+      this.httpService.get(`${process.env.VERCEL_BASE_URL}/v4/projects${teamId ? `?teamId=${teamId}` : ''}`, {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      })
+    );
+
+    return response.data.projects;
+  }
+
+  async getEnvKeys(command: SetupIntegrationCommand): Promise<{
+    privateKey: string;
+    clientKey: string;
+  }> {
+    const apiKeysResponse = await this.getApiKeyUsecase.execute(
+      GetApiKeysCommand.create({
+        environmentId: command.environmentId,
+        organizationId: command.organizationId,
+        userId: command.userId,
+      })
+    );
+
+    if (!Array.isArray(apiKeysResponse) || !apiKeysResponse[0].key) {
```

Comment:
The `apiKeysResponse` type says it's always an array, so the first check might not be needed, if it is then the type should be fixed. Also from what I see the API keys are per env not user, so the other part of the check might not be needed (to verify)?